File: src/main/java/com/github/zxh/classpy/classfile/bytecode/Iinc.java
package com.github.zxh.classpy.classfile.bytecode;

import com.github.zxh.classpy.classfile.ClassReader;
import com.github.zxh.classpy.classfile.jvm.Opcode;

public class Iinc extends Instruction {

    public Iinc(Opcode opcode, int pc) {
        super(opcode, pc);
    }
    
    @Override
    protected void readOperands(ClassReader reader) {
        int index = reader.readUnsignedByte();
        int _const = reader.readByte();
        setDesc(getDesc() + " " + index + ", " + _const);
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/bytecode/Iinc.kt
package com.github.zxh.classpy.classfile.bytecode

import com.github.zxh.classpy.classfile.reader.ClassReader

class Iinc(opcode: Opcode, pc: Int) : Instruction(opcode, pc) {

    override fun readOperands(reader: ClassReader) {
        val index = reader.readUnsignedByte()
        val const = reader.readByte()
        desc = "${opcode.name} $index, $const"
    }
    
}
